pytest/lib/utils.py

```diff
@@ -306,20 +307,35 @@ def compute_merkle_root_from_path(path, leaf_hash):
     return res
 
 
-def wait_for_blocks_or_timeout(node,
-                               num_blocks,
-                               timeout,
-                               callback=None,
-                               check_sec=1):
-    status = node.get_status()
-    start_height = status['sync_info']['latest_block_height']
-    max_height = 0
+def poll_blocks(node: LocalNode,
+                *,
+                timeout: float = 120,
+                poll_interval: float = 0.25,
+                **kw) -> typing.Iterable[typing.Tuple[int, str]]:
     started = time.time()
-    while max_height < start_height + num_blocks:
+    previous = -1
+    while True:
         assert time.time() - started < timeout
-        status = node.get_status()
-        max_height = status['sync_info']['latest_block_height']
-        if callback is not None:
-            if callback():
-                break
-        time.sleep(check_sec)
+        sync_info = node.get_status(**kw)['sync_info']
+        height = sync_info['latest_block_height']
+        if height != previous:
+            yield height, sync_info['latest_block_hash']
+            previous = height
+        time.sleep(poll_interval)
+
+
+def wait_for_blocks(node: LocalNode,
```

Comment:
let's add comments to all the util functions (so wait_for_blocks and. poll_blocks)

Reply:
I’ll do that as separate PR